src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -14,7 +14,7 @@ internal static class CompletionHelpers
     {
         private static readonly SearchValues<char> s_defaultCharsToCheck = SearchValues.Create("$`");
 
-        private static readonly SearchValues<char> s_escapeCharsToCheck = SearchValues.Create("$[]`");
+        private static readonly SearchValues<char> s_escapeGlobbingPathCharsToCheck = SearchValues.Create("$[]`");
```

Comment:
I wonder why do you name the chars "GlobbingPathChars"?

Interesting, is this used at all?

Reply:
So currently it is not used, but it is from the CompletionRequiresQuotes method. Most usage I see does not use it and keeps it `false`.

Perhaps we just remove it if not used instead of refacoring? Or we just decide to escape all special characters without this extra flag.

Reply:
Could you please look history of the file, maybe it was used in past?
It would be nice to understand why it is in code.

Reply:
@iSazonov So it seemed the original author wanted to be able to have the choice to escape globbed chars. To be honest I am not sure why it was ever flagged, there is no current code relying on it. But also making it the default could be done. Just not sure if it would introduce regression.

Reply:
I think for backwards compatibility and less risk of introducing regression, it seems fine to just rename this parameter to something more meaningful like in this PR. Currently `escape` on its own does not explain much what it is doing.

Reply:
We use WildcardPattern for filtering.
WildcardPattern uses brackets and backtick as special chars. So I'd expect we escape them for example for file paths but they escaped already so it is in another code path. So It is an open question.
Why should we escape `$`? Comment says about variables but variables are processed with another method `ContainsCharactersRequiringQuotes`. So it is an open question too.

@MartinGC94 Have you any thoughts?